### PR TITLE
[API 2.0] Add api paddle.reshape(x, shape, name)

### PR DESCRIPTION
**Pull request by liym27**
### PR types

Others
### PR changes

APIs
### Describe

Optimize api `paddle.reshape` for API2.0
image

### CODE REVIEW

File: python/paddle/fluid/layers/nn.py [hunk 1]
```diff
@@ -5959,7 +5959,7 @@ def autoincreased_step_counter(counter_name=None, begin=1, step=1):
     return counter
 
 
-def reshape(x, shape, actual_shape=None, act=None, inplace=False, name=None):
+def reshape(x, shape, actual_shape=None, act=None, name=None):
```
Comment (wangchaochaohu) [inline python/paddle/fluid/layers/nn.py]: nn.py函数不要修改

File: python/paddle/fluid/layers/nn.py [hunk 2]
```diff
@@ -6017,10 +6017,6 @@ def reshape(x, shape, actual_shape=None, act=None, inplace=False, name=None):
                                 This argument ``actual_shape`` will be removed in a future version. \
                                 Instructions for updating: ``actual_shape`` will be removed in future versions and replaced by ``shape``.
         act (str, optional): The non-linear activation to be applied to the reshaped input. Default None.
-        inplace(bool, optional): If ``inplace`` is True, the input and output of ``layers.reshape``
```
Comment (wangchaochaohu) [inline python/paddle/fluid/layers/nn.py:6020]: 不要删除

File: python/paddle/fluid/layers/nn.py [hunk 3]
```diff
@@ -6122,8 +6113,7 @@ def get_attr_shape(list_shape):
             actual_shape.stop_gradient = True
             inputs["Shape"] = actual_shape
 
-    out = x if inplace else helper.create_variable_for_type_inference(
```
Comment (wangchaochaohu) [inline python/paddle/fluid/layers/nn.py:6125]: nn.py 保持原样

**Comment by wangchaochaohu** [comment]
paddle.tensor.split(x,   num_or_sections, axis=0, name=None)


修改tensor下的split函数对齐最新的设计 同时单测完善了类型检查完善

**Comment by wangchaochaohu** [comment]
paddle.tensor.split(x,   num_or_sections, axis=0, name=None)


修改tensor下的split函数对齐最新的设计 同事单测完善了类型检查完善

File: python/paddle/fluid/layers/nn.py (hunk 1 above)
Comment (liym27) [reply python/paddle/fluid/layers/nn.py]: done

File: python/paddle/fluid/tests/unittests/test_reshape_op.py [hunk 4]
```diff
@@ -265,76 +273,107 @@ def test_1(self):
...
 
+    def test_paddle_api(self):
+        self._set_paddle_api()
+        self._test_api()
+
+    def test_fluid_api(self):
+        self._set_fluid_api()
+        self._test_api()
+
+    def test_imperative(self):
```
Comment (zhhsplendid) [inline python/paddle/fluid/tests/unittests/test_reshape_op.py:285]: Should imperative test also uses self.fill_constant, self.data, self.reshape to match your other test style?

Comment (liym27) [reply python/paddle/fluid/tests/unittests/test_reshape_op.py:285]: Thanks, done

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**